deepset_cloud_sdk/service/files_service.py

```diff
@@ -175,18 +177,76 @@ def _get_file_paths(paths: List[Path], recursive: bool = False) -> List[Path]:
-            elif os.path.isdir(path):
-                if recursive:
-                    for root, _, files in os.walk(path):
-                        for file in files:
-                            file_paths.append(Path(os.path.join(root, file)))
-                else:
-                    for file in os.listdir(path):
-                        file_path = os.path.join(path, file)
-                        if os.path.isfile(file_path):
-                            file_paths.append(Path(file_path))
+            elif recursive:
+                file_paths.extend([file_path for file_path in path.rglob("*")])
+            else:
+                file_paths.extend([file_path for file_path in path.glob("*")])
+        return file_paths
+
+    @staticmethod
+    def _validate_file_paths(file_paths: List[Path]) -> None:
+        """Validate a list of file paths.
+
+        This method validates the file paths and raises a ValueError if the file paths are invalid.
+        It also validates if there are meta files mapped to not existing raw files.
+
+        :param file_paths: A list of paths to upload.
+        :raises ValueError: If the file paths are invalid.
+        """
+        logger.info("Validating file paths and metadata")
+        allowed_suffixes = {".txt", ".json", ".pdf"}
+        for file_path in file_paths:
+            if not file_path.suffix.lower() in allowed_suffixes:
+                raise ValueError(f"Invalid file extension: {file_path.suffix}")
+            if file_path.suffix.lower() == ".json" and not str(file_path).endswith(".meta.json"):
+                raise ValueError(
+                    f"JSON files are only supported for meta files. Please make sure to name your files '<file_name>.meta.json'. Got {file_path.name}."
+                )
+        meta_file_names = list(
+            map(
+                lambda fp: os.path.basename(fp),
+                [file_path for file_path in file_paths if file_path.suffix.lower() == ".json"],
+            )
+        )
+        file_names = list(map(lambda fp: os.path.basename(fp), file_paths))
+        file_name_set = set(filter(lambda fn: not fn.endswith(".meta.json"), file_names))
+
+        not_mapped_meta_files = [
+            meta_file_name
+            for meta_file_name in meta_file_names
+            if not (meta_file_name.split(".meta.json")[0]) in file_name_set
+        ]
+
+        if len(not_mapped_meta_files) > 0:
+            raise ValueError(
+                f"Meta files without corresponding text files found: {not_mapped_meta_files}. "
+                "Please make sure that for each meta file there is a corresponding text file."
+                "The mapping needs to be done via file name '<file_name>' and '<file_name>.meta.json'. "
+                "For example: 'file1.txt' and 'file1.txt.meta.json'."
+            )
+
+    @staticmethod
+    def _preprocess_paths(paths: List[Path], spinner: Any = None, recursive: bool = False) -> List[Path]:
```

Comment:
Does the spinner have a typing yaspin.Spinner ?

Reply:
added